pkg/asset/installconfig/azure/client.go

```diff
@@ -211,3 +212,60 @@ func (c *Client) ListResourceIDsByGroup(ctx context.Context, groupName string) (
 	}
 	return res, nil
 }
+
+// GetVirtualMachineSku retrieves the restource SKU of a specified virtual machine in the given region.
+func (c *Client) GetVirtualMachineSku(ctx context.Context, name, region string) (azsku.ResourceSku, error) {
+	results, err := c.GetResourceSkus(ctx, "virtualMachines", region)
+	if err != nil {
+		return azsku.ResourceSku{}, err
```

Comment:
Can we change the return type to `*azsku.ResoureSku` so that we can return `nil` instead of an empty struct?

Reply:
Fixed in https://github.com/openshift/installer/commit/91402a22c47edd67c4b9802c4f2e47bfce3b3f39